src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,73 @@
+package codeu.controller;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.UUID;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.Before;
+import org.mockito.Mockito;
+
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import codeu.model.store.persistence.PersistentStorageAgent;
+
+import org.junit.Assert;
+
+import org.junit.Test;
+
+public class AdminServletTest {
+
+	private AdminServlet adminServlet;
+	private HttpServletRequest mockRequest;
+	private HttpServletResponse mockResponse;
+	private RequestDispatcher mockRequestDispatcher;
+	private ConversationStore mockConversationStore;
+	private MessageStore mockMessageStore;
+	private UserStore mockUserStore;
+	private PersistentStorageAgent mockPersistentStorageAgent;
+
+
+	@Before
+	public void setup() {
+		adminServlet = new AdminServlet();
+		mockRequest = Mockito.mock(HttpServletRequest.class);
+		mockResponse = Mockito.mock(HttpServletResponse.class);
+		mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+		Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/adminpage.jsp")).thenReturn(mockRequestDispatcher);
+		mockPersistentStorageAgent = Mockito.mock(PersistentStorageAgent.class);
+		
+		mockConversationStore = ConversationStore.getTestInstance(mockPersistentStorageAgent);
+		adminServlet.setConversationStore(mockConversationStore);
+
+		mockMessageStore = MessageStore.getTestInstance(mockPersistentStorageAgent);
+		adminServlet.setMessageStore(mockMessageStore);
+
+		mockUserStore = UserStore.getTestInstance(mockPersistentStorageAgent);
+		adminServlet.setUserStore(mockUserStore);
+	}
+
+	@Test
+	public void testDoGet() throws IOException, ServletException {
+		
+		int userCount = mockUserStore.getNumUsers();
```

Comment:
I think you want to test this by adding some users, some messges, and some conversations and testing that the expected value is set on the request rather than repeating the code from your implementation here.

I would literally add two users, then test
```
Mockito.verify(mockRequest).setAttribute("userCount", 2);
Mockito.verify(mockRequest).setAttribute("messageCount", 0);
Mockito.verify(mockRequest).setAttribute("conversationCount", 0);
```

then make a separate test for each attribute. The reason for this is that you want to make sure if you add two of each, you wouldn't know if users and messages were actually switched because they both are "2"